packages/backend-common/src/database/connectors/postgres.ts

```diff
@@ -110,9 +110,17 @@ export async function ensurePgDatabaseExists(
   dbConfig: Config,
   ...databases: Array<string>
 ) {
+  const adminDBConfig =
+    dbConfig.getOptionalConfig('connection.admin') || ({} as Config);
```

Comment:
Extended on my comment about using the same connection structure, I'm curious if you could re-use the something like `getPgConnectionConfig` here rather then all custom settings. This would make it more consistent across the default & plugin connections and likely simplify the createPgDatabaseClient call below as well.

Reply:
The option structure was literary POOMA, so yeah, I am all ears on a better config schema from the backstage maintainers.

But as for `getPgConnectionConfig`, I don't think there is an option to reuse it cause:
1. It gets the connection to make normal read write calls which has different creds
2. `createPgDatabaseClient` accepts config overrides which was how it is used currently. I am just adding extra params to override.

One other way to do the same would be to change `ensureDatabaseExists` in `connection.ts` by updating the `dbConfig` in that function with a similar logic. But I am not inclined to do it as this change seems to me to be very postgres specific.

Hope that gets my thoughts across. Do let me know your thoughts.

Reply:
I haven't pulled this code, but the `connection.admin` confuses me a bit when comparing to the `config.d.ts` changes. Can you provide an excerpt/redacted version of the `app-config.yaml` you have for your `backend.database` config?

Reply:
This is how my app config looks like
![image](https://user-images.githubusercontent.com/13748882/209564438-2561e78d-5585-43be-b415-e06024ca4d6a.png)
```yaml
backend:
  baseUrl: ${BASE_URL}
  listen:
    port: ${PORT}
  csp:
    connect-src: ['https:']
  database:
    client: pg
    pluginDivisionMode: schema
    ensureExists: ${DB_ENSURE_EXISTS}
    connection:
      host: ${DB_HOST}
      port: ${DB_PORT}
      user: ${DB_USER}
      password: ${DB_PASS}
      database: ${DB_NAME}
      ssl:
        sslmode: ${DB_SSL_MODE}
        rejectUnauthorized: ${DB_SSL_REJECT_UNAUTHORIZED}
```

Reply:
Hmm. Where is the usage of `admin`?

Reply:
Oh, well, I am not using it in my codebase yet. But I can give an example:
```yaml
backend:
  baseUrl: ${BASE_URL}
  listen:
    port: ${PORT}
  csp:
    connect-src: ['https:']
  database:
    client: pg
    pluginDivisionMode: schema
    ensureExists: ${DB_ENSURE_EXISTS}
    connection:
      host: ${DB_HOST}
      port: ${DB_PORT}
      user: ${DB_USER}
      password: ${DB_PASS}
      database: ${DB_NAME}
      ssl:
        sslmode: ${DB_SSL_MODE}
        rejectUnauthorized: ${DB_SSL_REJECT_UNAUTHORIZED}
    admin:
      user: ${ADMIN_USER}
      password: ${ADMIN_PASSWORD}
      database: ${ADMIN_DB}
```
The above is how things will look with the current structure. Yes, I realise that `admin` being on the same level as `connection` is weird but `connection` is untyped so I cannot put `admin` inside it.

Reply:
Based on the code that is reading `connection.admin` I was assuming you were nesting `admin` as a sibling with the other database settings default connection (full path `backend.database.connection.admin`. This will post issues

The connection settings are untyped because they are passed verbatim to `knex` and can be different per database client. The reason I'm suggesting making this `backend.database.adminConnection.connection` is make it possible to leverage this in other database clients in the future. It would allow the `postgres` settings to also use a connection string. Obviously tradeoffs, but for consistency I think using this model is better here, especially since plugin specific connections use the same pattern.

Reply:
Ugh...yeah, the code itself has an error in the part `dbConfig.getOptionalConfig('connection.admin')`. Should be `dbConfig.getOptionalConfig('admin')` instead. Now I get what you mean. Give me sometime, I will push a new commit with the changes and let me know what you think about it.